components/NameplateGroup.tsx

```diff
@@ -3,70 +3,136 @@ import Shape, {Polygon} from '../components/Shape';
 import styles from '../styles/lobby.module.scss';
 import {animatePolygon, animateNameplate} from '../utils/animations';
 
-export interface NameplateProps {
-  name: string;
+interface BaseAnimatedProps {
   shape: Polygon;
-  currentUser?: boolean;
+  highlight?: boolean;
   expanded?: boolean;
+  animate?: boolean;
+}
+
+interface AnimatedShapeProps extends BaseAnimatedProps {
+  shortText: string;
+  longText: string;
+}
+
+interface OverflowProps {
+  value: number;
+}
+
+export interface NameplateProps extends BaseAnimatedProps {
+  name: string;
+}
+
+interface NameplateGroupProps {
+  names: NameplateProps[];
+  expandCurrentUser?: boolean;
+  limit?: number;
 }
 
-export function Nameplate({name, shape, currentUser = false, expanded = false}: NameplateProps): JSX.Element {
+function AnimatedShape({
+  shortText,
+  longText,
+  shape,
+  highlight = false,
+  expanded = false,
+  animate = true,
+}: AnimatedShapeProps): JSX.Element {
   const [isHovered, setIsHovered] = useState(false);
   const [isFirstAnimation, setIsFirstAnimation] = useState(true);
 
   const containerRef = useRef(null);
   const polygonRef = useRef(null);
-  const shortNameRef = useRef(null);
-  const longNameRef = useRef(null);
+  const shortLabelRef = useRef(null);
+  const longLabelRef = useRef(null);
 
-  useEffect(() => {
-    if (isFirstAnimation) {
-      setIsFirstAnimation(false);
-      return;
+  useEffect(animate
+    ? () => {
+      if (isFirstAnimation) {
+        setIsFirstAnimation(false);
+        return;
+      }
+
+      animatePolygon({
+        target: containerRef.current,
+        polygonNode: polygonRef.current,
+        forwards: isHovered,
+        shape,
+      });
+      animateNameplate({
+        target: shortLabelRef.current,
+        subtarget: longLabelRef.current,
+        forwards: isHovered,
+      });
     }
+    : () => null, [isHovered]);
```

Comment:
i think maybe a useCallback might be better here...

```ts
const animateComponent = useCallback(() => {
  if (!animate) return;
  if (isFirstAnimation) { ... }
  // rest of animation code
}, [animate, isFirstAnimation]);
```

And then add to the `onHover` function:

```jsx
<div onMouseEnter{() => animateComponent()} />
```

not sure if this works tho... but it seems to clean things up?

Reply:
might be completely out of my mind tho haha

Reply:
I feel like `useEffect` feels more natural to use here. 

`useCallback` gets a little weird since you still need to tell it to either play the animation forward or not and the memoization on `animate` and `isFirstAnimation` doesn't seem too worth? Not sure though.

Reply:
i agree it feels more natural.. i guess I'm just not the fan of a state, and update hook on each `isHovered`. This is mostly because you have the `animate` property as a condition to access the animation component. So it makes sense to memorize the function?

Reply:
Getting rid of `useEffect` actually allows us to get rid of the `isFirstAnimation`, which does make it much cleaner